statement.py

```diff
@@ -330,16 +330,39 @@ def delete(cls, reconciliations):
             'from_account_bank_statement_line', False)
         if from_statement:
             return super(Reconciliation, cls).delete(reconciliations)
+
+        cls.check_bank_statement_lines(reconciliations)
+        return super(Reconciliation, cls).delete(reconciliations)
+
+    @classmethod
+    def check_bank_statement_lines(cls, reconciliations):
+        BankLines = Pool().get('account.bank.reconciliation')
+
+        moves = set()
         for reconciliation in reconciliations:
-            for line in reconciliation.lines:
-                for l in line.move.lines:
-                    for bank_line in line.bank_lines:
-                        if bank_line.bank_statement_line:
-                            raise UserError(gettext(
-                                'account_bank_statement_counterpart.reconciliation_cannot_delete',
-                                    reconciliation=reconciliation.rec_name,
-                                    line=bank_line.rec_name,
-                                    statement_line=(
-                                        bank_line.bank_statement_line.rec_name)
-                                    ))
-        super(Reconciliation, cls).delete(reconciliations)
+            moves = set(l.move for l in reconciliation.lines)
```

Comment:
I think you meant `moves |= set(l.move for l in reconciliation.lines)` ? Otherwise we'd be missing all moves except for the last reconciliation in the list!

Stilll, I'd rather write it in a single set comprehension:
`moves = set(line.move for reconciliation in reconciliations for line in reconciliation.lines)`